compiler/astalgo.nim

```diff
@@ -541,7 +542,37 @@ proc value(this: var DebugPrinter; value: PSym) =
 
   this.closeCurly
 
+proc builtinTypeSubstitution(this: var DebugPrinter; substitution: string) =
+  if this.useColor:
+    this.res.add backrefStyle
+  this.res.add substitution
+  if this.useColor:
+    this.res.add resetStyle
+
 proc value(this: var DebugPrinter; value: PType) =
+  # these shortcuts for builtin types are done before ``earlyExit`` because they are shorter that backreference links
+  if this.compressBuiltinTypes and value != nil:
+    case value.kind
+    of tyBool:     this.builtinTypeSubstitution "<bool>";    return
```

Comment:
use a template or macro to reduce repetitions

Reply:
No thanks, I prefer repetition here.

Reply:
``this.builtinTypeSubstitution(($value.kind)[2..^1].toLowerAscii)``

Reply:
@Araq, I know about this shorter code. But I really don't like unnecessary string allocations. Your version allocates two intermediate temporary strings, where my version creates none. Did you take that into account?

Reply:
you can cache these into a RT (or even CT) table and avoid allocs as well as code duplication.

Reply:
> Did you take that into account?

Yeah I did, allocations for debug code are irrelevant.